### DISCUSSION TITLE
Shared JSON property?

### DISCUSSION THREAD

**Discussion by GavinNL** [nlohmann/json#4319] [Ideas]
Hello, 

I currently have multiple json objects which need to share common property data. I tried searching the API/issues for something similar but wasn't able to find something. Would it be possible to have a property be a pointer to another json object, like so? 

```c++
auto external = std::make_shared<nlohmann::json>();
(*external)["world"] = 42;

nlohmann::json J;
J["hello"] = external;

assert(J["hello"]["world"] == 42);

(*external)["world"] = 52;
assert(J["hello"]["world"] == 52);

J["hello_bad"] = external->get(); // okay, but now must manage lifetime properly
```

**Comment by gregmarr** [comment]
There is nothing in the library that supports sharing a `json` object in multiple branches of the tree.
